### ISSUE TITLE
MarkerComposable

### ISSUE THREAD

**Issue by is-yusuf** [skiptools/skip-ui#60]
I was trying to add a custom marker to google maps using MarkerComposable.
The custom icon is generated from swiftUI and I cannot have the content of the MarkerComposable be the customPinView as I keep getting the error
`java.lang.IllegalStateException: The ComposeView was measured to have a width or height of zero. Make sure that the content has a non-zero size.`

My question is, what's the correct  way to use a swiftUI view that got transpiled to Kotlin view in a purely Kotlin file that I use in the /skip/ folder?

**Comment by aabewhite** [comment]
It might help to see your code. Normally to use a SwiftUI view in a Composable function, you just do `MyView(...).Compose()`. If you've moved from SwiftUI to Compose and now are going back to SwiftUI, pass in a ComposeContext derived from your parent ComposeView, as in:

```
ComposeView { ctx in 
   ... Compose stuff ...
   MyView().Compose(ctx.content())
   ... Compose stuff ...
}
```

**Comment by is-yusuf** [comment]
It's in /Sources/{AppName}/Resources/Resources.xcassets/custom_pin.imageset/Image.png
I tried setting width and height and didn't work either.

**Comment by aabewhite** [comment]
Does Image("custom_pin" ...) work in a SwiftUI view that is not embedded in Compose?

Can you show me how you're integrating MarkerComposable into your SwiftUI, or how you're displaying it in your app?

**Comment by is-yusuf** [comment]
Yes, Image(custom_pin"...) on it's own works if I write it in the swift part that gets transpiled. 
This is the code I have for the google map with custom markers.

```
GoogleMap(
        cameraPositionState = cameraPositionState,
        modifier = Modifier.fillMaxSize()
    ) {
        // Add a marker at the given location with "Hello World" as its content
        for (i in latitudeList.indices) {
            val markerPosition = LatLng(latitudeList[i], longitudeList[i])
            MarkerComposable(
                state = rememberMarkerState(position = markerPosition),
                title = "Marker at (${latitudeList[i]}, ${longitudeList[i]})",
                snippet = "Latitude: ${latitudeList[i]}, Longitude: ${longitudeList[i]}",
                onClick = {
                    openDialog.value = true
                    true
                }
            ) {
                Column {  CheckInPinView("10").Compose()}

            }
        }
        myModal(openDialog = openDialog)

    }
```

**Comment by aabewhite** [comment]
Sorry, I don't have any immediate ideas. You've tried the whole CheckInPinView and it works outside of the Compose function? How is the GoogleMap integrated into the rest of your app?

**Comment by is-yusuf** [comment]
Sorry for the late reply I got busy with another thing. 
The google map is integrated in a 

```
#if SKIP
ComposeView { ctx in
                    GoogleMapView(kotlin.arrayOf(Double(latitude)), kotlin.arrayOf(Double(longitude)))
                }
#endif

Where GoogleMapView.kt contains
```
GoogleMap(
        ...
    ) {
            MarkerComposable(
                state = rememberMarkerState(position = dummyMarkerLocation),
                  ...) {
                CustomPinView("10").Compose()
            }
        }
    }

```

The customPinView is defined in swift and gets transpiled.
 currently, customPinView shows only the text and fails to load the image when called from a kotlin file, but when called from a swift file it loads the image correctly.
